plugin/evm/mempool.go

```diff
@@ -259,16 +274,51 @@ func (m *Mempool) addTx(tx *Tx, force bool) error {
 	for utxoID := range utxoSet {
 		m.utxoSpenders[utxoID] = tx
 	}
+
+	m.bloom.Add(&GossipAtomicTx{Tx: tx})
+	reset, err := gossip.ResetBloomFilterIfNeeded(m.bloom, txGossipBloomMaxFilledRatio)
+	if err != nil {
+		return err
+	}
+
+	if reset {
+		log.Debug("resetting bloom filter", "reason", "reached max filled ratio")
+
+		for _, pendingTx := range m.txHeap.minHeap.items {
+			m.bloom.Add(&GossipAtomicTx{Tx: pendingTx.tx})
+		}
+	}
+
 	// When adding [tx] to the mempool make sure that there is an item in Pending
 	// to signal the VM to produce a block. Note: if the VM's buildStatus has already
 	// been set to something other than [dontBuild], this will be ignored and won't be
 	// reset until the engine calls BuildBlock. This case is handled in IssueCurrentTx
 	// and CancelCurrentTx.
 	m.newTxs = append(m.newTxs, tx)
 	m.addPending()
+
 	return nil
 }
 
+func (m *Mempool) Iterate(f func(tx *GossipAtomicTx) bool) {
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+
+	for _, item := range m.txHeap.maxHeap.items {
+		if !f(&GossipAtomicTx{Tx: item.tx}) {
+			return
+		}
+	}
+}
+
+func (m *Mempool) GetFilter() ([]byte, []byte, error) {
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+
+	bloom, err := m.bloom.Bloom.MarshalBinary()
+	return bloom, m.bloom.Salt[:], err
```

Comment:
I believe `ResetBloomFilterIfNeeded` reuses the `Salt` array - so wouldn't the returned slice be racy?

Reply:
In my head I for whatever reason thought I was doing a copy here. Updated to use a copy.